src/Microsoft.VisualStudio.ProjectSystem.Managed/MsBuildUtilities.cs

```diff
@@ -0,0 +1,172 @@
+            Requires.NotNull(evaluatedPropertyValue, nameof(evaluatedPropertyValue));
+            Requires.NotNullOrEmpty(propertyName, nameof(propertyName));
+
+            var property = GetOrAddProperty(project, propertyName);
+            StringBuilder newValue = new StringBuilder();
+            foreach (string value in GetPropertyValues(evaluatedPropertyValue, delimiter))
+            {
+                newValue.Append(value);
+                newValue.Append(delimiter);
+            }
+
+            newValue.Append(propertyValue);
+            property.Value = newValue.ToString();
+        }
+
+        /// <summary>
+        /// Renames a value inside a delimited property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="evaluatedPropertyValue">Original evaluated value of the property.</param>
+        /// <param name="propertyName">Property name.</param>
+        /// <param name="propertyValue">Value to remove from the property.</param>
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        /// <remarks>
+        /// If the property is not present it will be added. This means that the evaluated property
+        /// value came from one of the project imports.
+        /// </remarks>
+        public static void RemovePropertyValue(ProjectRootElement project, string evaluatedPropertyValue, string propertyName, string propertyValue, char delimiter = ';')
+        {
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(evaluatedPropertyValue, nameof(evaluatedPropertyValue));
+            Requires.NotNullOrEmpty(propertyName, nameof(propertyName));
+
+            var property = GetOrAddProperty(project, propertyName);
+            StringBuilder newValue = new StringBuilder();
+            foreach (string value in GetPropertyValues(evaluatedPropertyValue, delimiter))
+            {
+                if (value != propertyValue)
+                {
+                    newValue.Append(value);
+                    newValue.Append(delimiter);
+                }
+            }
+
+            property.Value = newValue.ToString().TrimEnd(delimiter);
+        }
+
+        /// <summary>
+        /// Renames a value inside a delimited property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="evaluatedPropertyValue">Original evaluated value of the property.</param>
+        /// <param name="propertyName">Property name.</param>
+        /// <param name="oldValue">Current property value.</param>
+        /// <param name="newValue">New property value.</param>
+        /// <param name="delimiter">Character used to delimit the property values.</param>
+        /// <remarks>
+        /// If the property is not present it will be added. This means that the evaluated property
```

Comment:
I would prefer option 2 since this indicates data mismatch. Either CPS or us should throw en error when we encounter this.